src/banner/Banner.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import uuidv4 from 'uuid/v4';
+
+import BannerRegistrationContext from './private/BannerRegistrationContext';
+import { BANNER_TYPES } from './private/utils';
+
+/* eslint-disable react/no-unused-prop-types */
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the banner.
+   */
+  action: PropTypes.element,
+  /**
+   * The pieces to populate a banner when type=`custom`.
+   */
+  custom: PropTypes.shape({
+    /**
+     * The title for the banner which will be bolded.
+     */
+    bannerTitle: PropTypes.string,
+    /**
+     * The class name used to set the icon in the banner.
+     */
+    icon: PropTypes.element,
+    /**
+     * Sets an author-defined class, to control the status bar color to be used.
+     */
+    colorClass: PropTypes.string,
+  }),
+  /**
+   * Nodes providing the message content for the banner. Can contain text and HTML.
+   */
+  description: PropTypes.node,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause
+   * the Dismiss button to be included on the banner.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `info`, `success`, `unsatisfied`, `unverified` or `custom`.
+   */
+  type: PropTypes.oneOf([
+    BANNER_TYPES.ALERT,
+    BANNER_TYPES.ERROR,
+    BANNER_TYPES.WARNING,
+    BANNER_TYPES.UNSATISFIED,
+    BANNER_TYPES.UNVERIFIED,
+    BANNER_TYPES.ADVISORY,
+    BANNER_TYPES.INFO,
+    BANNER_TYPES.SUCCESS,
+    BANNER_TYPES.CUSTOM,
+  ]).isRequired,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+const Banner = (props) => {
+  const bannerRegistration = React.useContext(BannerRegistrationContext);
+
+  React.useEffect(() => {
+    /**
+     * A unique identifier is generated for each Banner during construction. This will be used to
+     * uniquely register/unregister the banner with ancestor Banner Managers without requiring consumers to
+     * define unique identifiers themselves.
+     */
+    const uuid = uuidv4();
```

Comment:
Would it make sense to move this id into a ref outside the effect? Or do you want a new id with every registration?

Reply:
I created this [initially with used as a ref outside of the useEffect and was getting some lint warning](https://github.com/cerner/terra-application/pull/45/commits/7e46bcac48daa0cac03e65be0bc86f07c7b31aeb#diff-dc407e6487b82f8a1ebe8f58e5088673L63). I think our usage was fine to ignore the lint warning. It seems both accomplished the same thing, unique ID when rendered. Do you think we should disable the lint rule?

Reply:
You are right, the prop updsates triggers a re-render. Even when these are flattened, consumers may update the prop values and we would want to keep the same id. I will move back to be a ref.

Reply:
use ref: https://github.com/cerner/terra-application/pull/45/commits/bd880f41078aeda4a25edfab35fd79421c525c3a